### ISSUE TITLE
Issue for S608 seems off

### ISSUE THREAD

**Issue by colin99d** [charliermarsh/ruff#2778]
Running `ruff .` gives a weird warning, see below:

binaries/xgboost/sklearn.py:133:15: S608 Possible SQL injection vector through string-based query construction: "f\"\\n    max_depth :  Optional[int]\\n        Maximum tree depth for base learners.\\n    max_leaves :\\n        Maximum number of leaves; 0 indicates no limit.\\n    max_bin :\\n        If using histogram-based algorithm, maximum number of bins per feature\\n    grow_policy :\\n        Tree growing policy. 0: favor splitting at nodes closest to the node, i.e. grow\\n        depth-wise. 1: favor splitting at nodes with highest loss change.\\n    learning_rate : Optional[float]\\n        Boosting learning rate (xgb\'s \\\"eta\\\")\\n    verbosity : Optional[int]\\n        The degree of verbosity. Valid values are 0 (silent) - 3 (debug).\\n    objective : {_SklObjective}\\n        Specify the learning task and the corresponding learning objective or\\n        a custom objective function to be used (see note below).\\n    booster: Optional[str]\\n        Specify which booster to use: gbtree, gblinear or dart.\\n    tree_method: Optional[str]\\n        Specify which tree method to use.  Default to auto.  If this parameter is set to\\n        default, XGBoost will choose the most conservative option available.  It\'s\\n        recommended to study this option from the parameters document :doc:`tree method\\n        </treemethod>`\\n    n_jobs : Optional[int]\\n        Number of parallel threads used to run xgboost.  When used with other\\n        Scikit-Learn algorithms like grid search, you may choose which algorithm to\\n        parallelize and balance the threads.  Creating thread contention will\\n        significantly slow down both algorithms.\\n    gamma : Optional[float]\\n        (min_split_loss) Minimum loss reduction required to make a further partition on a\\n        leaf node of the tree.\\n    min_child_weight : Optional[float]\\n        Minimum sum of instance weight(hessian) needed in a child.\\n    max_delta_step : Optional[float]\\n        Maximum delta step we allow each tree\'s weight estimation to be.\\n    subsample : Optional[float]\\n        Subsample ratio of the training instance.\\n    sampling_method :\\n        Sampling method. Used only by `gpu_hist` tree method.\\n          - `uniform`: select random training instances uniformly.\\n          - `gradient_based` select random training instances with higher probability when\\n            the gradient and hessian are larger. (cf. CatBoost)\\n    colsample_bytree : Optional[float]\\n        Subsample ratio of columns when constructing each tree.\\n    colsample_bylevel : Optional[float]\\n        Subsample ratio of columns for each level.\\n    colsample_bynode : Optional[float]\\n        Subsample ratio of columns for each split.\\n    reg_alpha : Optional[float]\\n        L1 regularization term on weights (xgb\'s alpha).\\n    reg_lambda : Optional[float]\\n        L2 regularization term on weights (xgb\'s lambda).\\n    scale_pos_weight : Optional[float]\\n        Balancing of positive and negative weights.\\n    base_score : Optional[float]\\n        The initial prediction score of all instances, global bias.\\n    random_state : Optional[Union[numpy.random.RandomState, int]]\\n        Random number seed.\\n\\n        .. note::\\n\\n           Using gblinear booster with shotgun updater is nondeterministic as\\n           it uses Hogwild algorithm.\\n\\n    missing : float, default np.nan\\n        Value in the data which needs to be present as a missing value.\\n    num_parallel_tree: Optional[int]\\n        Used for boosting random forest.\\n    monotone_constraints : Optional[Union[Dict[str, int], str]]\\n        Constraint of variable monotonicity.  See :doc:`tutorial </tutorials/monotonic>`\\n        for more information.\\n    interaction_constraints : Optional[Union[str, List[Tuple[str]]]]\\n        Constraints for interaction representing permitted interactions.  The\\n        constraints must be specified in the form of a nested list, e.g. ``[[0, 1], [2,\\n        3, 4]]``, where each inner list is a group of indices of features that are\\n        allowed to interact with each other.  See :doc:`tutorial\\n        </tutorials/feature_interaction_constraint>` for more information\\n    importance_type: Optional[str]\\n        The feature importance type for the feature_importances\\\\_ property:\\n\\n        * For tree model, it\'s either \\\"gain\\\", \\\"weight\\\", \\\"cover\\\", \\\"total_gain\\\" or\\n          \\\"total_cover\\\".\\n        * For linear model, only \\\"weight\\\" is defined and it\'s the normalized coefficients\\n          without bias.\\n\\n    gpu_id : Optional[int]\\n        Device ordinal.\\n    validate_parameters : Optional[bool]\\n        Give warnings for unknown parameter.\\n    predictor : Optional[str]\\n        Force XGBoost to use specific predictor, available choices are [cpu_predictor,\\n        gpu_predictor].\\n    enable_categorical : bool\\n\\n        .. versionadded:: 1.5.0\\n\\n        .. note:: This parameter is experimental\\n\\n        Experimental support for categorical data.  When enabled, cudf/pandas.DataFrame\\n        should be used to specify categorical data type.  Also, JSON/UBJSON\\n        serialization format is required.\\n\\n    feature_types : FeatureTypes\\n\\n        .. versionadded:: 1.7.0\\n\\n        Used for specifying feature types without constructing a dataframe. See\\n        :py:class:`DMatrix` for details.\\n\\n    max_cat_to_onehot : Optional[int]\\n\\n        .. versionadded:: 1.6.0\\n\\n        .. note:: This parameter is experimental\\n\\n        A threshold for deciding whether XGBoost should use one-hot encoding based split\\n        for categorical data.  When number of categories is lesser than the threshold\\n        then one-hot encoding is chosen, otherwise the categories will be partitioned\\n        into children nodes. Also, `enable_categorical` needs to be set to have\\n        categorical feature support. See :doc:`Categorical Data\\n        </tutorials/categorical>` and :ref:`cat-param` for details.\\n\\n    max_cat_threshold : Optional[int]\\n\\n        .. versionadded:: 1.7.0\\n\\n        .. note:: This parameter is experimental\\n\\n        Maximum number of categories considered for each split. Used only by\\n        partition-based splits for preventing over-fitting. Also, `enable_categorical`\\n        needs to be set to have categorical feature support. See :doc:`Categorical Data\\n        </tutorials/categorical>` and :ref:`cat-param` for details.\\n\\n    eval_metric : Optional[Union[str, List[str], Callable]]\\n\\n        .. versionadded:: 1.6.0\\n\\n        Metric used for monitoring the training result and early stopping.  It can be a\\n        string or list of strings as names of predefined metric in XGBoost (See\\n        doc/parameter.rst), one of the metrics in :py:mod:`sklearn.metrics`, or any other\\n        user defined metric that looks like `sklearn.metrics`.\\n\\n        If custom objective is also provided, then custom metric should implement the\\n        corresponding reverse link function.\\n\\n        Unlike the `scoring` parameter commonly used in scikit-learn, when a callable\\n        object is provided, it\'s assumed to be a cost function and by default XGBoost will\\n        minimize the result during early stopping.\\n\\n        For advanced usage on Early stopping like directly choosing to maximize instead of\\n        minimize, see :py:obj:`xgboost.callback.EarlyStopping`.\\n\\n        See :doc:`Custom Objective and Evaluation Metric </tutorials/custom_metric_obj>`\\n        for more.\\n\\n        .. note::\\n\\n             This parameter replaces `eval_metric` in :py:meth:`fit` method.  The old one\\n             receives un-transformed prediction regardless of whether custom objective is\\n             being used.\\n\\n        .. code-block:: python\\n\\n            from sklearn.datasets import load_diabetes\\n            from sklearn.metrics import mean_absolute_error\\n            X, y = load_diabetes(return_X_y=True)\\n            reg = xgb.XGBRegressor(\\n                tree_method=\\\"hist\\\",\\n                eval_metric=mean_absolute_error,\\n            )\\n            reg.fit(X, y, eval_set=[(X, y)])\\n\\n    early_stopping_rounds : Optional[int]\\n\\n        .. versionadded:: 1.6.0\\n\\n        Activates early stopping. Validation metric needs to improve at least once in\\n        every **early_stopping_rounds** round(s) to continue training.  Requires at least\\n        one item in **eval_set** in :py:meth:`fit`.\\n\\n        The method returns the model from the last iteration (not the best one).  If\\n        there\'s more than one item in **eval_set**, the last entry will be used for early\\n        stopping.  If there\'s more than one metric in **eval_metric**, the last metric\\n        will be used for early stopping.\\n\\n        If early stopping occurs, the model will have three additional fields:\\n        :py:attr:`best_score`, :py:attr:`best_iteration` and\\n        :py:attr:`best_ntree_limit`.\\n\\n        .. note::\\n\\n            This parameter replaces `early_stopping_rounds` in :py:meth:`fit` method.\\n\\n    callbacks : Optional[List[TrainingCallback]]\\n        List of callback functions that are applied at end of each iteration.\\n        It is possible to use predefined callbacks by using\\n        :ref:`Callback API <callback_api>`.\\n\\n        .. note::\\n\\n           States in callback are not preserved during training, which means callback\\n           objects can not be reused for multiple training sessions without\\n           reinitialization or deepcopy.\\n\\n        .. code-block:: python\\n\\n            for params in parameters_grid:\\n                # be sure to (re)initialize the callbacks before each run\\n                callbacks = [xgb.callback.LearningRateScheduler(custom_rates)]\\n                xgboost.train(params, Xy, callbacks=callbacks)\\n\\n    kwargs : dict, optional\\n        Keyword arguments for XGBoost Booster object.  Full documentation of parameters\\n        can be found :doc:`here </parameter>`.\\n        Attempting to set a parameter via the constructor args and \\\\*\\\\*kwargs\\n        dict simultaneously will result in a TypeError.\\n\\n        .. note:: \\\\*\\\\*kwargs unsupported by scikit-learn\\n\\n            \\\\*\\\\*kwargs is unsupported by scikit-learn.  We do not guarantee\\n            that parameters passed via this argument will interact properly\\n            with scikit-learn.\\n\""

**Comment by charliermarsh** [comment]
Maybe we should just omit the string itself.

**Comment by charliermarsh** [comment]
Or truncate.
